src/perl.c

```diff
@@ -2026,13 +2062,19 @@ static void perl_log (int level, const char *msg,
 
 	/* Lock the base thread if this is not called from one of the read threads
 	 * to avoid race conditions with c_ithread_create(). See
-	 * https://github.com/collectd/collectd/issues/9 for details. */
-	if (aTHX == perl_threads->head->interp)
+	 * https://github.com/collectd/collectd/issues/9 for details.
+	 * Additionally check, if we are called from perl interpreter.
+	 * Maybe PTHREAD_MUTEX_RECURSIVE mutex type will be more appropriate?
```

Comment:
I can add additional variable and pthread_mutexattr_init(), pthread_mutexattr_settype() and pthread_mutexattr_destroy() required calls.  But I have no much desire to repeat full testing which was already done. Maybe leave this untouched at this moment? What do you think?